ert_shared/storage/storage_api.py

```diff
@@ -116,134 +112,144 @@ def _calculate_misfit(
         return {"value": misfit, "sign": sign, "obs_index": obs_index}
 
     def get_response(self, ensemble_id, response_name, filter):
-        with self._rdb_api as rdb_api, self._blob_api as blob_api:
-            bundle = rdb_api.get_response_bundle(
-                response_name=response_name, ensemble_id=ensemble_id
-            )
-            if bundle is None:
-                return None
-
-            observation_links = bundle.observation_links
-            responses = bundle.responses
-            univariate_misfits = {}
-            for resp in responses:
-                resp_values = list(blob_api.get_blob(resp.values_ref).data)
-                univariate_misfits[resp.realization.index] = {}
-                for link in observation_links:
-                    observation = link.observation
-                    obs_values = list(blob_api.get_blob(observation.values_ref).data)
-                    obs_stds = list(blob_api.get_blob(observation.stds_ref).data)
-                    obs_data_indexes = list(
-                        blob_api.get_blob(observation.data_indexes_ref).data
-                    )
-                    misfits = []
-                    for obs_index, obs_value in enumerate(obs_values):
-                        misfits.append(
-                            self._calculate_misfit(
-                                obs_value,
-                                resp_values,
-                                obs_stds,
-                                obs_data_indexes,
-                                obs_index,
-                            )
+        bundle = self._rdb_api.get_response_bundle(
```

Comment:
I think we need to keep the usage of context managers to allow for the db session to be closed

Reply:
I'll just explain my reasoning here, correct me if I'm wrong. 

The reference in question is a member of the object. It seems to me that this is a confusing and wrong usage of context managers. The code was a leftover from a time when _rdb_api and _blop_api were created on demand, while that is no longer the case. It is now the entire StorageApi that is a resourse to be used in a "with" statement, and the _api members are closed when it is closed. I also got crashes in code as it was, since the "with" statement closed the _api resources when done, but the StorageApi used them again.

Reply:
that is wrong 
currently you can use the rdb_api and blob_api if you provide a connection - it will open and close the session within a context

if you are using storage_api - the functions should use rdb and blob api context managers to retrieve session ( and close those ) within a function. the same goes for the extraction api. 

if we not open and close sessions and connections towards the database properly - we really fast end up in connection and session errors which is the whole reason for making it as it is today. 

we could discuss if there could be done some improvements - i don't think what we have today is the perfect solution - but by removing context-managers we will end up in with a solution not working.

Reply:
There is something I'm misunderstanding. The rdp_api and blob_api enter function does nothing, but the exit function closes the connection. This leaves the object in a unusable state I would presume, and yet it is there as a member of the StorageApi still. It seems completely sufficient to me to use the StorageApi in a with statement, and when done, close it, which will close also the rdb and blob apis.

Reply:
the rdb_api and blob_api has this 
``` 
    def __init__(self, connection):
        self._session = Session(bind=connection)

    def __enter__(self):
        return self

    def __exit__(self, type, value, traceback):
        self.close()
```

on enter it initalize a session and bind that to the connection sent in, on exit - it releases the session

the storage_api  (and exctraction api) context managers using the db urls to create a connection and holds on to that for the lifetime of the objects and with the connections initalizes and holds on to the rdb and blob apis. 

it ended up this way because we had problems with the connections not beeing closed and objects beeing requested from one thread but created in another. Thus we had to fix the session stuff and connection stuff to be inside the objects. 

It is not 100% all good, there should probably be some improvements and refactorings of this structure - but i don't think those should belong to this PR.

Reply:
I did this because I was getting crashes, and this fixed it. Not sure if this code still requires it, but I'll check. I don't see how the change made here actually changes the  behavior in any usecase that wouldn't have crashed either way.

Should I create another issue for this and let this PR wait if it is required?

Reply:
need to dig a bit into this and understand what happens and why it fail for you since it's been running smoothly for a while now with everything else. Could be this is perfectly fine  - but if it is - then probably more should change. at least that is why i wanted you to create an issue if there is an bug. if you have found a bug it would be kind of stupid just fixing it at this point and leave the rest behind. i can try to investigate a bit this evening / tomorrow and see if i can understand 👍

Reply:
I removed the removal of the with with the latest commit. Doesn't seem like this is an issue any more for this PR.